sample/hello_world/requirements.txt

```diff
@@ -0,0 +1 @@
+requests
```

Comment:
not used/imported
Maybe it's intentional for testing dependency packaging (but then the code should at least import it)?

Reply:
Let's leave it in as you say it checks that packaging works correctly. I am happy not import it since the sample application is what `samlocal init` generates.

It might be good to create a test that runs `samlocal init` first before deploying to check that that part works, but I think this is a more useful test as it ensures our endpoint overrides are working correctly.

Reply:
Agree that packaging testing makes a lot of sense 👍 

I would still use the import in the code because packing can also be done wrong; it's not critical here because we are not testing sam itself ...

Reply:
I think you're right, I have had enough trouble with pipelines not invoking sam correctly which misses dependencies. I agree that it's not our job to test sam, but we should test that we are invoking it correctly and I'd prefer to have an end-to-end test that's more realistic than not.

Reply:
Fully agree 👍 

IMHO, such an end-to-end test should also check whether the package is usable because Python dependencies need to follow a directory structure to work properly.